### PR TITLE
Custom serial selection

### PR DESCRIPTION
**Pull request by pablogs9**
Custom serial selection using an integer

### CODE REVIEW

File: rmw_microxrcedds_c/src/config.h.in [hunk 1]
```diff
@@ -14,11 +14,9 @@
     #define RMW_UXRCE_MAX_TRANSPORT_MTU UXR_CONFIG_UDP_TRANSPORT_MTU
     #define RMW_UXRCE_DEFAULT_UDP_IP "@RMW_UXRCE_DEFAULT_UDP_IP@"
     #define RMW_UXRCE_DEFAULT_UDP_PORT "@RMW_UXRCE_DEFAULT_UDP_PORT@"
-#elif defined(MICRO_XRCEDDS_SERIAL)
-    #define RMW_UXRCE_MAX_TRANSPORT_MTU UXR_CONFIG_UDP_TRANSPORT_MTU
-    #define RMW_UXRCE_DEFAULT_SERIAL_DEVICE "@RMW_UXRCE_DEFAULT_SERIAL_DEVICE@"
-#elif defined(MICRO_XRCEDDS_CUSTOM)
+#elif defined(MICRO_XRCEDDS_SERIAL) || defined(MICRO_XRCEDDS_CUSTOM)
```
Comment (julibert) [inline rmw_microxrcedds_c/src/config.h.in]: Maybe we should rename `MICRO_XRCE_CUSTOM` as `MICRO_XRCE_CUSTOM_SERIAL`.

File: rmw_microxrcedds_c/src/rmw_init.c [hunk 2]
```diff
@@ -218,8 +218,12 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
...
   printf("UDP mode => ip: %s - port: %s\n", context_impl->connection_params.agent_address, context_impl->connection_params.agent_port);
 #elif defined(MICRO_XRCEDDS_CUSTOM)
-  if (!uxr_init_serial_transport(&context_impl->transport, &context_impl->serial_platform, 0, 0, 1))
-  {
+  int pseudo_fd = 0;
+  if (strlen(options->impl->connection_params.serial_device) > 0){
+    pseudo_fd = atoi(options->impl->connection_params.serial_device);
+  }
+  
+  if (!uxr_init_serial_transport(&context_impl->transport, &context_impl->serial_platform, pseudo_fd, 0, 1)){
```
Comment (julibert) [inline rmw_microxrcedds_c/src/rmw_init.c]: How does it work previously? We didn't set the fd.

Comment (pablogs9) [reply rmw_microxrcedds_c/src/rmw_init.c]: Well previously the custom transport opens and uses a certain UART interface and it ignores the file descriptor. Now we are enabled to pass a file descriptor to the custom transport, this way in FreeRTOS+Olimex you can select between UART3, 6 or 1.

File: rmw_microxrcedds_c/src/config.h.in (hunk 1 above)
Comment (pablogs9) [reply rmw_microxrcedds_c/src/config.h.in]: Done

**Comment by pablogs9** [comment]
DYW, done

**Pull request merged**